isopropylcyanide/Jwt-Spring-Security-JPA
# Bug/password reset verification

## Summary
| PR Status  | Type  | Impact level |
| :---: | :---: | :---: |
| Bug  | Feature | High |

## Description
This PR adds / changes the following

- Password reset token request takes "user email" and validate it
- Password reset token has a "claimed", "active" field and updated at, created at field
- Validate that password reset token is marked as **claimed** and marks all active tokens for the user as inactive
- Junit5 Migration for existing tests + New tests (enhancement)
- Update README 

## Motivation & Context
- There are **3 major bugs** in password reset flow. #57 
 
- Another issue is on observability. #58 We can't currently track when a token was created, when it was updated. 





## How was this tested
- Unit tests have been added.
- Verified that the existing logouts functionality works
![image](https://user-images.githubusercontent.com/12872673/139558353-6acbe64d-1470-41e9-a3b5-a6f0f0926872.png)

- Verify using the same token results in an error
![image](https://user-images.githubusercontent.com/12872673/139558357-7911ca94-8918-4e58-b51d-cb1e15e679c8.png)

- Verify using someone else's token results in an error 
![image](https://user-images.githubusercontent.com/12872673/139558431-96304d05-4564-4eeb-9b4b-94514b71ba8b.png)

- Verify "claim" and "active" are being set and unset correctly.
![image](https://user-images.githubusercontent.com/12872673/139558448-2c8304b5-d7a0-4547-8b5f-fc8c946de339.png)
